pom.xml

```diff
@@ -38,10 +38,10 @@
         </pluginRepository>
     </pluginRepositories>
     <properties>
-        <jenkins.version>1.642.3</jenkins.version>
+        <jenkins.version>2.7.3</jenkins.version>
```

Comment:
Various plugin/core bumps are so we can do `properties(...)` and `parameters: [booleanParam(...)]` and have fairly current (for core 2.7.3, at least) dependencies for tests. I basically lifted all the dependency versions straight from Declarative, which is also on 2.7.3.